src/stactools/nrcan_spot_ortho/cog.py

```diff
@@ -0,0 +1,218 @@
+import os
+from tempfile import TemporaryDirectory
+import pystac
+from pystac.extensions.eo import EOExtension
+from pystac.extensions.projection import ProjectionExtension
+from stactools.nrcan_spot_ortho.stac_templates import image_types
+from stactools.nrcan_spot_ortho.geobase_ftp import GeobaseSpotFTP
+from stactools.nrcan_spot_ortho.stac_templates import (spot_bands, spot_pan,
+                                                       proj_epsg)
+from stactools.nrcan_spot_ortho.utils import (CustomStacIO, download_from_ftp,
+                                              call, get_existing_paths, unzip,
+                                              upload_to_s3, file_exists)
+from urllib.parse import urlparse
+import rasterio
+
+pystac.StacIO.set_default(CustomStacIO)
+
+
+def cogify(input_path, output_path, overwrite, existing_cog_paths):
+    """COGify a geotiff at input_path to a cloud optimized geotiff at output_path.
+    """
+    print(f"COGifying {os.path.basename(input_path)}")
+    failure = False
+    parsed = urlparse(output_path)
+
+    if (not overwrite) and (output_path in existing_cog_paths):
+        print(f"Skipping {os.path.basename(output_path)}, already COGified.")
+
+    elif parsed.scheme == "s3":
+        with TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, os.path.basename(output_path))
+            failure = call([
+                'gdal_translate', '-of', 'COG', '-co', 'compress=deflate',
+                input_path, tmp_path
+            ])
```

Comment:
Not required, but there is a `cogify` command in `stactools.core.utils`: https://github.com/stac-utils/stactools/blob/a2e541e18d1d50689c5a0a5bb224bccbcbdf6819/src/stactools/core/utils/convert.py#L8-L20